src/business_logic/execution/os_usage.rs

```diff
@@ -1,26 +1,81 @@
 use std::collections::HashMap;
 
-#[derive(Debug, Clone, Default)]
+use cairo_rs::vm::runners::cairo_runner::ExecutionResources;
+
+use crate::definitions::transaction_type::TransactionType;
+
+use super::execution_errors::ExecutionError;
+
+#[derive(Debug, Clone)]
 pub struct OsResources {
     _execute_syscalls: HashMap<String, ExecutionResources>,
-    _execute_txs_inner: HashMap<TransactionType, ExecutionResources>,
+    execute_txs_inner: HashMap<TransactionType, ExecutionResources>,
 }
 
 // TODO: add the hash maps that are in os_resources.json in cairo-lang
 
 impl OsResources {
-    pub fn new() -> Self {
+    pub fn default() -> Self {
+        let execute_txs_inner: HashMap<TransactionType, ExecutionResources> = HashMap::from([
+            (
+                TransactionType::InvokeFunction,
+                ExecutionResources {
+                    n_steps: 2839,
+                    n_memory_holes: 0,
+                    builtin_instance_counter: HashMap::from([
+                        ("pedersen_builtin".to_string(), 16),
+                        ("range_check_builtin".to_string(), 70),
+                    ]),
+                },
+            ),
+            (
+                TransactionType::Declare,
+                ExecutionResources {
+                    n_steps: 2336,
+                    n_memory_holes: 0,
+                    builtin_instance_counter: HashMap::from([
+                        ("pedersen_builtin".to_string(), 15),
+                        ("range_check_builtin".to_string(), 57),
+                    ]),
+                },
+            ),
+            (
+                TransactionType::Deploy,
+                ExecutionResources {
+                    n_steps: 0,
+                    n_memory_holes: 0,
+                    builtin_instance_counter: HashMap::new(),
+                },
+            ),
+            (
+                TransactionType::DeployAccount,
+                ExecutionResources {
+                    n_steps: 3098,
+                    n_memory_holes: 0,
+                    builtin_instance_counter: HashMap::from([
+                        ("pedersen_builtin".to_string(), 23),
+                        ("range_check_builtin".to_string(), 74),
+                    ]),
+                },
+            ),
+        ]);
+
         OsResources {
-            ..Default::default()
+            _execute_syscalls: HashMap::new(),
+            execute_txs_inner,
         }
     }
 }
 
 pub fn get_additional_os_resources(
     _syscall_counter: HashMap<String, u64>,
-    _tx_type: TransactionType,
-) -> ExecutionResources {
-    todo!();
+    tx_type: &TransactionType,
+) -> Result<ExecutionResources, ExecutionError> {
+    let os_resources = OsResources::default();
+
+    Ok(os_resources
+        .execute_txs_inner
+        .get(tx_type)
+        .ok_or_else(|| ExecutionError::NoneTransactionType(tx_type.clone(), os_resources.clone()))?
+        .clone())
```

Comment:
@pefontana i think this function is not doing the same as:
https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/business_logic/execution/os_usage.py#L29

Can you clarify?

Reply:
Maybe will be a good idea to ask if this is still important or a legacy staff

Reply:
It does not seem like a legacy thing. But please ask the Starkware team :)

Reply:
Also, how are you getting the expected added resourses? Are you checking this in your test?

Reply:
Kind of,
In the test, the `let result = internal_invoke_function._apply_specific_concurrent_changes(&mut state, &StarknetGeneralConfig::default()).unwrap();` returns a TransactionExecutionInfo that has an empty ExecutionResources: `ExecutionResources { n_steps: 0, n_memory_holes: 0, builtin_instance_counter: {} }, events: [], l2_to_l1_messages: [], storage_read_values: [], accesed_storage_keys: {}, internal_calls: [] })`
I didn't assert it because I am not sure what the expected output should be